plugins/inventory/vmware_vm_inventory.py

```diff
@@ -395,211 +485,255 @@ def parse(self, inventory, loader, path, cache=True):
+                for cust_value in vm_obj.obj.customValue:
+                    properties[[y.name for y in field_mgr if y.key == cust_value.key][0]] = cust_value.value
+
+            # Tags
+            if tags_info:
+                # Add virtual machine to appropriate tag group
+                vm_mo_id = vm_obj.obj._GetMoId()
+                vm_dynamic_id = DynamicID(type='VirtualMachine', id=vm_mo_id)
+                attached_tags = [tags_info[tag_id] for tag_id in tag_association.list_attached_tags(vm_dynamic_id)]
+                properties['tags'] = attached_tags
+
+            # Build path
+            if with_path:
+                path = []
+                parent = vm_obj.obj.parent
+                while parent:
+                    path.append(parent.name)
+                    parent = parent.parent
+                path.reverse()
+                properties['path'] = "/".join(path)
+
+            host_properties = to_nested_dict(properties)
+            host = self._get_hostname(host_properties, hostnames)
+
+            if host not in hostvars:
+                hostvars[host] = host_properties
+                self._populate_host_properties(host_properties, host)
+
+        return hostvars
+
+    def _populate_host_properties(self, host_properties, host):
         # Load VM properties in host_vars
-        vm_properties = self.get_option('properties') or []
-
-        field_mgr = self.pyv.content.customFieldsManager.field
-
-        for vm_prop in vm_properties:
-            if vm_prop == 'customValue':
-                for cust_value in vm_obj.customValue:
-                    self.inventory.set_variable(current_host,
-                                                [y.name for y in field_mgr if y.key == cust_value.key][0],
-                                                cust_value.value)
-            else:
-                vm_value = self.pyv._get_object_prop(vm_obj, vm_prop.split("."))
-                self.inventory.set_variable(current_host, vm_prop, vm_value)
+        with_path = self.get_option('with_path')
+        with_nested_properties = self.get_option('with_nested_properties')
+
+        self.inventory.add_host(host)
+
+        # Use constructed if applicable
+        strict = self.get_option('strict')
+        # Composed variables
+        self._set_composite_vars(self.get_option('compose'), host_properties, host, strict=strict)
+        # Complex groups based on jinja2 conditionals, hosts that meet the conditional are added to group
+        self._add_host_to_composed_groups(self.get_option('groups'), host_properties, host, strict=strict)
+        # Create groups based on variable values and add the corresponding hosts to it
+        self._add_host_to_keyed_groups(self.get_option('keyed_groups'), host_properties, host, strict=strict)
+
+        # group by parents
+        # TODO: This should be a feature of Constractable:
+        #   keyed_groups:
+        #       - parent_group: '{{ path.split("/") }}'
```

Comment:
parent_group is templated so it seems like you could do this already if a key is provided - what isn't working?

Reply:
I think the current `parent_group` implementation is not recursive. My suggestion is to allow it to accept a list of string as an input and generate group tree based on it. For example:

```py
path = "/parent3/parent2/parent1/host_vm"
```

```yaml
 keyed_groups:
   - parent_group: '{{ path.split("/") }}'
```

That would generate the following:
```json
{
    "Parent3": {
        "children": [
            "Parent2"
        ]
    },
    "Parent2": {
        "children": [
            "Parent1"
        ]
    },
    "Parent1": {
        "hosts": [
            "host_vm"
        ]
    }
},
"all": {
    "children": [
        "Parent3",
        ...
    ]
}
```